### PR TITLE
[STK-320] Add kubegen support

### PR DESCRIPTION
**Pull request by waseem-h**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -5,4 +5,7 @@ gem "oj", "3.5.1"
 gem "fluent-plugin-elasticsearch"
 gem "fluent-plugin-kubernetes_metadata_filter"
 gem "ffi"
-gem "fluent-plugin-systemd"
\ No newline at end of file
+gem "fluent-plugin-systemd"
+gem "fluent-plugin-concat", "2.1.0"
```
Comment (rasheedamir) [inline Gemfile:9]: why do we need this plugin? I remember it creates unnecessary issues

Comment (waseem-h) [reply Gemfile:9]: We haven't removed the use of this plugin yet. Once it is removed from the config, I'll remove it from here as well

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. `docker pull docker.io/stakater/fluentd-debian9-kubernetes:1.2.2-0.0.7`

File: Gemfile (hunk 1 above)
Comment (rasheedamir) [reply Gemfile:9]: but in this repo there isn't any config which uses this plugin or I am missing something?

File: Dockerfile [hunk 2]
```diff
@@ -37,10 +42,14 @@ COPY Gemfile* /fluentd/
 COPY ./conf/fluent.conf /fluentd/etc/
 COPY ./conf/systemd.conf /fluentd/etc/
 COPY ./conf/kubernetes.conf /fluentd/etc/
+COPY ./fluent.conf.tpl /fluentd/etc/template/
```
Comment (rasheedamir) [inline Dockerfile]: we should rather call it **dynamic-conf** or something else

**Review by rasheedamir — CHANGES_REQUESTED** [review]
As discussed there should be:

1. one parent conf file which includes other conf files
2. kube-gen generated conf file must have per app block with id

File: conf/fluent.conf [hunk 3]
```diff
@@ -1,26 +1,4 @@
...
-     flush_thread_count 8
-     flush_interval 5s
-     chunk_limit_size 2M
-     queue_limit_length 32
-     retry_max_interval 30
-     retry_forever true
-   </buffer>
-</match>
\ No newline at end of file
+@include kube-gen.conf
```
Comment (rasheedamir) [inline conf/fluent.conf:4]: you forgot to add **elasticsearch-out.conf** ? @waseem-h

**Review by rasheedamir — CHANGES_REQUESTED** [review]

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. `docker pull docker.io/stakater/fluentd-debian9-kubernetes:1.2.2-0.0.8`

File: conf/fluent.conf (hunk 3 above)
Comment (waseem-h) [reply conf/fluent.conf:4]: It is included in kube-gen.conf.tpl

Comment (waseem-h) [reply conf/fluent.conf:4]: Moved it to fluentd.conf

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. `docker pull docker.io/stakater/fluentd-debian9-kubernetes:1.2.2-0.0.9`

**Review by rasheedamir — APPROVED** [review]
Looks good

**Pull request merged**